transport/grpc/handler.go

```diff
@@ -283,3 +289,29 @@ func handlerErrorToGRPCError(err error, responseWriter *responseWriter) error {
 	}
 	return status.Error(grpcCode, message)
 }
+
+// toGRPCError converts errors to gRPC status errors. gRPC status errors are
+// returned as is.
+//
+// This MUST NOT be used for coverting YARPC error details to gRPC error
+// details. Use toGRPCErrorWithDetails instead.
+func toGRPCError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	// if this is not a yarpc error, return the error
+	// this will result in the error being a grpc-go error with codes.Unknown
+	if !yarpcerrors.IsStatus(err) {
+		return err
+	}
+	// we now know we have a yarpc error
+	yarpcStatus := yarpcerrors.FromError(err)
+	message := yarpcStatus.Message()
+	grpcCode, ok := grpcerrorcodes.YARPCCodeToGRPCCode[yarpcStatus.Code()]
+	// should only happen if grpcerrorcodes.YARPCCodeToGRPCCode does not cover all codes
+	if !ok {
+		grpcCode = codes.Unknown
+	}
+	return status.Error(grpcCode, message)
```

Comment:
nit: message only used once. No need to assign to a variable. `return status.Error(grpcCode. yarpcStatus.Message())`

Reply:
This block was moved as-is from stream.go, so I'd rather not touch it in this PR.